### ISSUE TITLE
Can we get a man page added?

### ISSUE THREAD

**Issue by brd** [cperciva/bsdec2-image-upload#12]

**Comment by cperciva** [comment]
Sure!  Want to write one?

**Comment by brd** [comment]
I'll take a stab, but some questions..

1. What does `--vhd` actually mean?  I was able to upload an image and it created an AMI, but now I want to replace the image that AMI uses so I can track `main`.  Could it be used for this?
2. What does `--arm64` do?  Should it be made more abstract?
3. I would like to use bsdec2-image-upload in other tooling, what do you think about a new option to output the image name to file or something?

**Comment by cperciva** [comment]
> 1. What does `--vhd` actually mean?  I was able to upload an image and it created an AMI, but now I want to replace the image that AMI uses so I can track `main`.  Could it be used for this?

The `--vhd` option says "this disk image is in VHD format rather than a raw disk image".  It also changes the method used for uploading the image since EC2 has special code for handling VHD images.

> 2. What does `--arm64` do?  Should it be made more abstract?

Tells EC2 that this is an arm64 image.  The default, for backwards compatibility reasons, is amd64.  We could make it a general `--platform` option but I don't think there's much point; we either have amd64 or arm64 now and for the forseeable future.

> 3. I would like to use bsdec2-image-upload in other tooling, what do you think about a new option to output the image name to file or something?

Redirect standard output?  But if you think a separate option would be useful, I have no objection to having one.

**Comment by brd** [comment]
Thanks for the explanations.

Redirecting works with `env WITH_CLOUDWARE=y make -C release cw-ec2` and running it myself.

**Comment by brd** [comment]
Do the artifacts left in the s3 bucket need to be preserved?  The manifest.xml and ~500 partXXX files?

**Comment by cperciva** [comment]
No.  I recommend using an S3 bucket with a lifecycle rule which deletes everything after 24 hours.

**Comment by cperciva** [comment]
@brd Are you still interested in writing a man page?
